File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/RemoteLoader.java
package com.example.try_gameengine.remotecontroller;

import com.example.try_gameengine.framework.BitmapUtil;
import com.example.try_gameengine.framework.Config;

/**
 * @author irons
 *
 */
public class RemoteLoader {
	
	private UpKey upKey;
	private DownKey downKey;
	private RightKey rightKey;
	private LeftKey leftKey;
	private RemoteControl remoteControl;
	
	public RemoteLoader() {
		load();
	}
	
	private void load(){
		remoteControl = new RemoteControl();
		
		rightKey = new RightKey(BitmapUtil.rightKey, Config.currentScreenWidth-BitmapUtil.rightKey.getWidth(), Config.currentScreenHeight-BitmapUtil.rightKey.getHeight(), 1, false);
		rightKey.setEnableMultiTouch(true);
		RightKeyCommandPressDown upKeyCommandPressDown = new RightKeyCommandPressDown(rightKey);
		RightKeyCommandPressUp upKeyCommandPressUp = new RightKeyCommandPressUp(rightKey);
		leftKey = new LeftKey(BitmapUtil.leftKey, 0, Config.currentScreenHeight-BitmapUtil.leftKey.getHeight(), 1, false);
		leftKey.setEnableMultiTouch(true);
		LeftKeyCommandPressDown leftKeyCommandPressDown = new LeftKeyCommandPressDown(leftKey);
		LeftKeyCommandPressUp leftKeyCommandPressUp = new LeftKeyCommandPressUp(leftKey);
		
		remoteControl.setCommand(0, upKeyCommandPressDown, upKeyCommandPressUp);
		remoteControl.setCommand(1, leftKeyCommandPressDown, leftKeyCommandPressUp);
	}
	
	public UpKey getUpKey(){
		return upKey;
	}
	
	public DownKey getDownKey(){
		return downKey;
	}
	
	public LeftKey getLeftKey(){
		return leftKey;
	}
	
	public RightKey getRightKey(){
		return rightKey;
	}
	
	public RemoteControl getRemoteControl(){
		return remoteControl;
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/RemoteLoader.kt
package com.example.try_gameengine.remotecontroller

import com.example.try_gameengine.framework.BitmapUtil
import com.example.try_gameengine.framework.Config

/**
 * @author irons
 // */
class RemoteLoader {
    private val upKey: UpKey? = null
    private val downKey: DownKey? = null
    private var rightKey: RightKey? = null
    private var leftKey: LeftKey? = null
    private var remoteControl: RemoteControl? = null

    init {
        load()
    }

    private fun load() {
        remoteControl = RemoteControl()

        rightKey = RightKey(
            BitmapUtil.rightKey!!,
            Config.currentScreenWidth - BitmapUtil.rightKey!!.getWidth(),
            Config.currentScreenHeight - BitmapUtil.rightKey!!.getHeight(),
            1,
            false
        )
        val right = rightKey!!
        right.isEnableMultiTouch = true
        val upKeyCommandPressDown = RightKeyCommandPressDown(right)
        val upKeyCommandPressUp = RightKeyCommandPressUp(right)
        leftKey = LeftKey(
            BitmapUtil.leftKey!!,
            0f,
            Config.currentScreenHeight - BitmapUtil.leftKey!!.getHeight(),
            1,
            false
        )
        val left = leftKey!!
        left.isEnableMultiTouch = true
        val leftKeyCommandPressDown = LeftKeyCommandPressDown(left)
        val leftKeyCommandPressUp = LeftKeyCommandPressUp(left)

        remoteControl!!.setCommand(0, upKeyCommandPressDown, upKeyCommandPressUp)
        remoteControl!!.setCommand(1, leftKeyCommandPressDown, leftKeyCommandPressUp)
    }

    fun getUpKey(): UpKey? {
        return upKey
    }

    fun getDownKey(): DownKey? {
        return downKey
    }

    fun getLeftKey(): LeftKey? {
        return leftKey
    }

    fun getRightKey(): RightKey? {
        return rightKey
    }

    fun getRemoteControl(): RemoteControl {
        return remoteControl!!
    }
}
